### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by OstrichDowneyJr**
It just works!

### CODE REVIEW

File: Project2/main.c
Comment (KidziaK) [inline Project2/main.c:1]: **issue**: Don't add files not related with the task/issue to the PR.

File: project1/CLI_strings/Dummy.txt
Comment (KidziaK) [inline project1/CLI_strings/Dummy.txt:1]: **suggestion**: the test coverage is quite poor, consider adding additional tests with certain edge cases.

File: project1/CLI_strings/main.c [hunk 1]
```diff
@@ -0,0 +1,35 @@
+#include "main.h"
+
+
+int main(int argc, char *argv[]) {
+
+	///*handling main arguments*/
+	if (argc != 3) {
```
Comment (KidziaK) [inline project1/CLI_strings/main.c]: **nit**: Really nice error handling. Two things to consider though: 
1) The comment is not really necessary it's usually better to leave the code to explain itself (you can add non-doc comments when there is a design choice that might not be clear to the reader, or when there is some complicated algorithm involved, and you want to provide a reference).
2) To make this error handling even better, you could also print the number of arguments the user provided or even the exact arguments they have provided, e.g., 
`printf("Wrong number of arguments, expected 2, got %i\n", argc);`

File: project1/CLI_strings/main.c [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+		printf("Wrong number of arguments, expected 2\n");
+		return EXIT_FAILURE;
+	}
+	FILE* file;
+	/*variables declaraion*/
+	char read_buff[BUFF_SIZE];
+
+	/*getting file to program's memory*/
+	if ((file = fopen(argv[1], "r")) == NULL) {
```
Comment (KidziaK) [inline project1/CLI_strings/main.c]: **nit:** Avoid declaring/assigning variables inside _if_ statement condition. It's often confusing for the reader and there is no real benefit.

File: project1/CLI_strings/main.c [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+		printf("File opening error!\n");
```
Comment (KidziaK) [inline project1/CLI_strings/main.c:37]: **suggestion**: You can improve the error message by printing the path to the invalid/non-existing file.

File: project1/CLI_strings/main.c [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+		return EXIT_FAILURE;
+	}
+	fgets(read_buff, BUFF_SIZE,file);
+	fclose(file);
+	/*spliting strings to separate lines*/
+	char* token;
```
Comment (KidziaK) [inline project1/CLI_strings/main.c]: **nit**: name **token** is confusing. `strtok()` returns a pointer to a string section not a single character or token. Consider renaming it to something like **current_section**.

File: project1/CLI_strings/main.c [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+	/*to nie dziala, nie umiem zapisac tego wskaznika*/
+	token= strtok(read_buff,argv[2]);
```
Comment (KidziaK) [inline project1/CLI_strings/main.c]: **suggestion**: Instead of argv[1] and argv[2] it's a good idea to parse them at the beginning of the script and give them meaningful names, e.g., 
`char* path_to_txt = argv[1];`
`char* delimiters = argv[2];`

File: project1/CLI_strings/main.c [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+	while (token != NULL) {
+		printf("%s\n", token);
+		token = strtok(NULL,argv[2]); /*tu tez*/
```
Comment (KidziaK) [inline project1/CLI_strings/main.c]: **issue**: Remember to remove debug code and comments (_/\*tu tez\*/_) before PR.

File: project1/CLI_strings/main.h
Comment (KidziaK) [inline project1/CLI_strings/main.h:1]: **suggestion**: There is nothing wrong with this file, however I would suggest to put everything in a single file unless you reuse that code in other parts of your system, or when your source file becomes to large and hard to maintain.

File: project1/CLI_strings/main.c
Comment (KidziaK) [inline project1/CLI_strings/main.c:1]: **observation**: In the current form the file does not meet the following requirement: _- Delimiter does not have to be a single character._ However it's not that important for this project, so I'll let this one slide :D Just remove this note from the README.

**Review by KidziaK — CHANGES_REQUESTED** [review]
